### PR TITLE
FEAT: Layer corner types

### LINKED ISSUE #17: Layer style - round corners with options to specify corner type

### PR DESCRIPTION
**Pull request by jakubpetrik**
resolves #17

### CODE REVIEW

File: Example/Styles/Sources/AppDelegate.swift [hunk 1]
```diff
@@ -46,7 +46,7 @@ let body = TextStyle(
 )
 
 let rounded = LayerStyle(
-        .cornerRadius(10),
+        .roundCorners([.topLeft, .bottomRight], radius: 10),
```
Comment (radimhalfar) [inline Example/Styles/Sources/AppDelegate.swift]: How about to keep both options ? most cases are fully rounded, huh ?

File: Styles/Classes/LayerStyle.swift [hunk 2]
```diff
@@ -2,23 +2,68 @@
 
 import UIKit
 
+extension UIRectCorner {
+    @available(iOS 11.0, *)
+    var maskedCorners: CACornerMask {
+        var mask: CACornerMask = []
+        if self.contains(.topLeft) {
```
Comment (radimhalfar) [inline Styles/Classes/LayerStyle.swift]: is self required here ?

File: Styles/Classes/LayerStyle.swift [hunk 3]
```diff
@@ -2,23 +2,68 @@
...
+        }
+        return mask
+    }
+}
+
 public final class LayerStyle: NSObject {
     public enum Property {
         case borderColor(UIColor)
         case borderWidth(CGFloat)
-        case cornerRadius(CGFloat)
```
Comment (radimhalfar) [inline Styles/Classes/LayerStyle.swift]: i would keep both options whereas `cornerRadius` renamed to `roundCorners(CGFloat)`

File: Example/Styles/Sources/AppDelegate.swift (hunk 1 above)
Comment (jakubpetrik) [reply Example/Styles/Sources/AppDelegate.swift]: for all rounded corners you create following 
```
.roundCorners(.allCorners, radius: 10)
```

**Pull request merged**